datafusion/sql/src/planner.rs

```diff
@@ -189,8 +189,27 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
                 query,
                 with_options,
                 ..
-            } if columns.is_empty() && with_options.is_empty() => {
-                let plan = self.query_to_plan(*query, &mut HashMap::new())?;
+            } if with_options.is_empty() => {
+                let mut plan = self.query_to_plan(*query, &mut HashMap::new())?;
+
+                if !columns.is_empty() {
+                    plan = LogicalPlanBuilder::from(plan.clone())
+                        .project(
+                            plan.schema().fields().iter().zip(columns.into_iter()).map(
+                                |(field, ident)| {
+                                    col(field.name()).alias(&normalize_ident(&ident))
+                                },
+                            ),
+                        )
+                        .map_err(|e| {
+                            context!("Failed to apply alias to inline projection.\n", e)
+                        })?
+                        .build()
```

Comment:
This is fine and no need to change this but just an FYI really, you can avoid having multiple `map_err` like this:

```suggestion
                        .and_then(|builder| builder.build())
```